packages/web-components/src/components/va-file-input/va-file-input.tsx

```diff
@@ -0,0 +1,169 @@
+import { 
+  Component, 
+  Element, 
+  Host, 
+  h, 
+  Prop, 
+  Build, 
+  forceUpdate,
+  Fragment,
+  Event,
+  EventEmitter
+} from '@stencil/core';
+import i18next from 'i18next';
+
+if (Build.isTesting) {
+  // Make i18next.t() return the key instead of the value
+  i18next.init({ lng: 'cimode' });
+}
+
+@Component({
+  tag: 'va-file-input',
+  styleUrl: 'va-file-input.css',
+  shadow: true,
+})
+
+export class VaFileInput {
+  @Element() el: HTMLElement;
+
+  /**
+   * The label for the button input.
+   */
+  @Prop() label?: string;
+
+  /**
+   * The name for the input element.
+   */
+  @Prop() name?: string;
+
+  /**
+   * The text displayed on the button (required)
+   */
+  @Prop() buttonText!: string;
```

Comment:
Should this be required? I see you're providing a fallback value of `i18next.t('Upload file')` for the button.

Reply:
@andresriveratoro It does seem redundant now (the fallback request was part of the design discussion). I'll clean that up.